torchgeo/transforms/transforms.py

```diff
@@ -73,3 +73,100 @@ def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
             sample["boxes"] = sample["boxes"].to(boxes_dtype)
 
         return sample
+
+
+class NCrop(Module):
+    """Take repeated random crops from a tile.
+
+    This is usefule to create *num_patches_per_tile* random
+    patches from a tile, to have a manageable input dimension
+    size for various models.
+
+    .. versionadded:: 0.4
+    """
+
+    def __init__(
+        self, patch_size: Union[Tuple[int, int], int], num_patches_per_tile: int
+    ) -> None:
+        """Initialize a new instance of NCrop.
+
+        Args:
+            patch_size: Size of random patch from image and mask (height, width)
+            num_patches_per_tile: number of patches to randomly crop from a tile
+        """
+        super().__init__()
+        self.num_patches_per_tile = num_patches_per_tile
+        self.patch_size = patch_size
+
+    def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        """Perform random cropping on a tile to create patches.
+
+        Args:
+            sample: the input
+
+        Returns:
+            stacked random patches
+        """
+        images, masks = [], []
+        for i in range(self.num_patches_per_tile):
+            crop = K.RandomCrop(self.patch_size, p=1.0)
```

Comment:
This can result in possibly overlapping tiles. Not necessarily a deal breaker, but something that Kornia is likely to point out.